docxtpl/template.py

```diff
@@ -351,6 +351,19 @@ def render_properties(
             rendered = template.render(context)
             setattr(self.docx.core_properties, prop, rendered)
 
+    def render_footnotes(
+            self, context: Dict[str, Any], jinja_env: Optional[Environment] = None
+    ) -> None:
+        if jinja_env is None:
+            jinja_env = Environment()
+
+        for section in self.docx.sections:
+            for part in section.part.package.parts:
+                if part.content_type == 'application/vnd.openxmlformats-officedocument.wordprocessingml.footnotes+xml':
+                    xml = self.patch_xml(part.blob.decode('utf8'))
```

Comment:
`part.blob` somehow becomes `str` in the next loop body if we have multiple footnotes in the template.

```
Traceback (most recent call last):
  File "/Users/cmiki/Projects/sq-docxgen/src/templating/main.py", line 76, in <module>
    tpl.render(context, jinja_env)
  File "/Users/cmiki/Projects/python-docx-template/docxtpl/template.py", line 499, in render
    self.render_footnotes(context, jinja_env)
  File "/Users/cmiki/Projects/python-docx-template/docxtpl/template.py", line 363, in render_footnotes
    xml = self.patch_xml(part.blob.decode('utf8'))
                         ^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'decode'. Did you mean: 'encode'?
```

here's a duct-tape approach (which works on my template):
```suggestion
                    xml = self.patch_xml(part.blob.decode('utf-8') if type(part.blob) is bytes else part.blob)
```

Reply:
Thanks for this, I believe my test only processes one footnote, so it didn't catch this.

For now I will apply your suggestion, but replace the `type(part.blob) is bytes` pattern with `isinstance(part.blob, bytes)`.

I will also try to narrow down what causes this in the `python-docx` library, which I believe might be causing this type confusion.

Reply:
Awesome. Thank you for this patch!